File: app/src/main/java/br/com/concrete/desafio_android/domain/RepositoryResponse.java
package br.com.concrete.desafio_android.domain;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by Daivid
 * To ilustrate how the Gson works
 */
public class RepositoryResponse implements Serializable {

    private int total_count = 0;
    private boolean incomplete_results = true;
    private List<Repository> items = new ArrayList<Repository>();
    private String message = new String();
    private String documentation_url = new String();

    public int getTotal_count() {
        return total_count;
    }

    public void setTotal_count(int total_count) {
        this.total_count = total_count;
    }

    public boolean isIncomplete_results() {
        return incomplete_results;
    }

    public void setIncomplete_results(boolean incomplete_results) {
        this.incomplete_results = incomplete_results;
    }

    public List<Repository> getItems() {
        return items;
    }

    public void setItems(List<Repository> items) {
        this.items = items;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getDocumentation_url() {
        return documentation_url;
    }

    public void setDocumentation_url(String documentation_url) {
        this.documentation_url = documentation_url;
    }
}

File: app/src/main/java/com/concrete/andresdavid/desafioandroid/model/RepositoryResponse.kt
package com.concrete.andresdavid.desafioandroid.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


class RepositoryResponse() {
    @SerializedName("total_count")
    @Expose
    private val totalCount: Int = 0

    @SerializedName("incomplete_results")
    @Expose
    private var incompleteResults: Boolean = false

    @SerializedName("items")
    @Expose
    var items: List<Repository> = emptyList()
}